docs/usage-guide/topics/ch04-connection.md

```diff
@@ -1,6 +1,6 @@
 # TLS Connections
 
-Users will need to create a `s2n_connection` struct to store all of the state necessary for a TLS connection. Call `s2n_connection_new()` to create a new server or client connection. Call `s2n_connection_free()` to free the memory allocated for this struct when no longer needed.
+Users will need to create a `s2n_connection` struct to store all of the state necessary for a TLS connection. One `s2n_connection` must be created for each TCP stream. Call `s2n_connection_new()` to create a new server or client connection. Call `s2n_connection_free()` to free the memory allocated for this struct when no longer needed.
```

Comment:
We've sort of lost the original point of the PR now, which was to provide an overview of "s2n-tls software architecture". In my mind, that looks something like: "In general, s2n-tls works by operating on s2n_connections. A user should create a connection, associate a s2n_config with that connection, and then calling s2n_negotiate on the connection until the TLS handshake completes. Then users can call s2n_send/s2n_recv on the connection object to send and receive application data."

It's fine to provide that sort of general overview in the beginning of the connections.md file.